packages/@aws-cdk/aws-ec2/lib/nat.ts

```diff
@@ -45,6 +61,11 @@ export abstract class NatProvider {
    * Called by the VPC to configure NAT
    */
   public abstract configureNat(options: ConfigureNatOptions): void;
+
+  /**
+   * Return list of gateways spawned by the provider
+   */
+  public abstract onlineGateways(): IGateway[];
```

Comment:
Why `onlineGateways`, and not just `gateways` or `gatewaysCreated`?

Reply:
Class `NetProvider` already have a method `gateway`.   Semantic of `gateway` and `gateways` is different.  I'd like to ensure that developers would not be confused by almost identical methods. I would rename `gateway` to `NatProvider().withGateway()` but it potentially breaks Public API :-(

Reply:
added `configuredGateways()` method